CONTRIBUTING.md

```diff
@@ -0,0 +1,244 @@
+- `file` archive file name
+- `format` archive format (zip | tar | gztar | bztar | xztar)
+
+For replacement operations:
+- `$type: regex_replace`
+- `file` name of file where replacement must be executed
+- `pattern` string or regexp ([learn more](https://docs.python.org/2/library/re.html))
+- `replacement` replacement string
+- `count` (*optional*)  maximum number of pattern occurrences to be replaced
+
+**`conversion_to_onnx_args`** (*optional*)
+
+List of onnx conversion parameters, see `model_optimizer_args` for details. Applicable for Caffe2\* and PyTorch\* frameworks.
+
+**`model_optimizer_args`**
+
+Conversion parameters (learn more in [Model conversion](#model-conversion) section) is specified in this section, e.g.:
+```
+  - --input=data
+  - --mean_values=data[127.5]
+  - --scale_values=data[127.5]
+  - --reverse_input_channels
+  - --output=prob
+  - --input_model=$conv_dir/googlenet-v3.onnx
+```
+> **NOTE:**  no need to specify `framework`, `data_type`, `model_name` and `output_dir` parameters since them are deduced automatically.
+
+**`framework`**
+
+Framework of original model (`caffe`, `dldt`, `mxnet`, `pytorch`, `tf`).
+
+**`license`**
+
+Path to model's license.
+
+----
+*After this step you will obtain **model.yml** file*
+
+## Model conversion
+
+Deep Learning Inference Engine (IE) supports models in Intermediate Representation (IR) format. A model from any supported framework can be converted to IR using Model Optimizer tool included in OpenVINO&trade; package. Find more information about conversion in [Model Optimizer Developer Guide](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html). After successful conversion you will get model in IR format `*.xml` representing net graph and `*.bin` containing net parameters. 
+
+> **NOTE 1**: due to OpenVINO&trade; paradigms, image pre-processing parameters (mean and scale) should be built into converted model to simplify model usage.
+
+> **NOTE 2**: due to OpenVINO&trade; paradigms, if model input is a color image, color channel order should be `BGR`.
+
+*After this step you`ll get **conversion parameters** for Model Optimizer.*
+
+## Demo
+
+The demo shows the main idea of model inference using IE. If your model solves one of the tasks supported by Open Model Zoo, find appropriate demo from [demos](https://docs.openvinotoolkit.org/latest/_demos_README.html) or sample from [samples](https://docs.openvinotoolkit.org/latest/_docs_IE_DG_Samples_Overview.html).
+
+If appropriate demo or sample are absent, you must provide your own demo (C++ or Python). Demos are required to support the following keys:
+
+-    `-i "<input>"`             Required. Input to process.
+-    `-m "<path>"`              Required. Path to an .xml file with a trained model.
+-    `-d "<device>"`            Optional. Target device for model inference. Default is CPU.
+-    `-no_show`                 Optional. Do not visualize inference results.
```

Comment:
@Daniil-Osokin noted, that for python we should use `-` instead of `_` as word separator, so `no_show` should be `no-show` for python. What do you think?

Reply:
@Wovchena I've watched almost all Python demos and didn't find `no-show` or `no_show`. But almost all demos use `_` instead of `-`.  So I can mention your point in the guide, but it will lead to demo edit.

Reply:
I can update demos when the PR is ready.

Reply:
While I prefer dashes in options, it it IMHO not worth it to change existing demos at this point. I think we should just standardize on underscores.

Reply:
What about allowing new demos following any of two styles and we point it in contribution guide?

Reply:
Why should we do that? The fewer styles, the better.

Reply:
The other style is more native for python :)

Reply:
It's not exactly foreign to C++ either... But as I've said, I don't think it's worth it to change the convention at this point.

Reply:
OK